src/main/java/graphql/schema/SchemaTransformer.java

```diff
@@ -34,6 +34,55 @@
 @PublicApi
 public class SchemaTransformer {
 
+    /**
+     * Different ways to traverse a schema object during transformation.
+     *
+     * ALL - traverse all top level types (query, mutations, additional types etc)
+     * QUERY_ONLY - traverse Query only, basically ignoring schema elements that are not referenced in the query.
+     */
+    public enum TraversalType {
+        ALL,
+        QUERY_ONLY;
```

Comment:
Naming is hard - I would call this `OPERATION_TYPES_ONLY` which comes from

https://graphql.github.io/graphql-spec/June2018/#OperationType

That is the operation types of the schema.  The use of the name query is misleading because we have Mutation / Query and Subscription

Reply:
Good point - oversight on my part. Will update.

Reply:
As discussed this needs to include mutations and subscriptions (and in the supporting code